### DISCUSSION TITLE
separate audio for each sentence with corresponding file name

### DISCUSSION THREAD

**Discussion by omega3** [pndurette/gTTS#432] [Q&A] [answered]
I use gtts-cli installed with pipx. I have a spreadsheet with two columns, first column is a sentence number, second column contains sentences. Like this:
01 This is sentence one.
02 This is sentence two.

For each sentence I want to produce separate audio file.

Can you help me and tell me how to use gtts-cli to produce audio output files that as a name have number of the sentence, so that I know which file contains which sentence?

It may be exported as csv.

I know I need to use:

```
gtts-cli -f speak.txt --output test.mp3
```

1. Do I need to have a separate file for each single sentence and loop this command or can I have all sentences in one file?

2. If loop is needed how can I do it?

3.  Do I need some breaks like a second or so, between each query to not overload google, to not cause some blockade?

**Comment by lacorrep** [comment, marked as answer]
To create a single file for each sentence (`01.mp3`, `02.mp3`...), save your spreadsheet in CSV then run this Python script:
```python
import csv
from gtts import gTTS
# mport time # if you want to add pauses because you are worried

filename = 'my_sentences.csv' # name of the spreadsheet file in CSV format

with open(filename, 'r') as csvfile:
	reader = csv.reader(csvfile)
	for row in reader:
		sentence_number, sentence = row
		gTTS(sentence, lang='en', lang_check=False, slow=False).save(sentence_number+'.mp3')
		# time.sleep(1) # wait 1 second if you are worried
```

If you know how to code, you could also loop the command line you wrote using the BASH language. If you want to create a single file, I gave the code in my question #312.

**Reply by omega3** [reply]
Thank you very much. It works flawlessly.
I have additional question: can this be used with programs installed by pipx? For example [piper](https://github.com/rhasspy/piper)?

Something like:
`from gtts-cli import gTTS`
In other words: is possible to import not something installed system-wide but something that is run from pipx?

I also asked similar question on [pipx discussions](https://github.com/pypa/pipx/discussions/1086).

And I changed my question in #312 If you could have another look, please.

**Reply by lacorrep** [reply]
CLI means "command-line interface": it only makes sense to use it in a terminal, not in a script (`.py` file).
I don't know pipx but maybe it works like any other [virtual environment](https://docs.python.org/3/library/venv.html), so I guess it should work.
